### ISSUE TITLE
unable to enter toolbox; "no matching entries in passwd file"

### ISSUE THREAD

**Issue by miabbott** [debarshiray/toolbox#196]
Hit this today with toolbox 0.10 and podman 1.4.0

```
$ toolbox list

$ toolbox -v create
toolbox: resolved absolute path for /usr/bin/toolbox to /usr/bin/toolbox
toolbox: TOOLBOX_PATH is /usr/bin/toolbox
toolbox: checking if 'podman system migrate' exists
toolbox: migration not needed: 1.4.0 is unchanged
toolbox: Fedora generational core is f30
toolbox: base image is fedora-toolbox:30
toolbox: container is fedora-toolbox-30
toolbox: checking value /var/run/.heim_org.h5l.kcm-socket (Stream) of property Listen in sssd-kcm.socket
toolbox: parsing value /var/run/.heim_org.h5l.kcm-socket (Stream) of property Listen in sssd-kcm.socket
toolbox: checking if 'podman create' supports --dns=none and --no-hosts
toolbox: 'podman create' supports --dns=none and --no-hosts
toolbox: looking for image localhost/fedora-toolbox:30
toolbox: looking for image registry.fedoraproject.org/f30/fedora-toolbox:30
Image required to create toolbox container.
Download registry.fedoraproject.org/f30/fedora-toolbox:30 (500MB)? [y/N]: y
toolbox: pulling image registry.fedoraproject.org/f30/fedora-toolbox:30
Trying to pull registry.fedoraproject.org/f30/fedora-toolbox:30...Getting image source signatures
Copying blob 4c4fc0337a75 done
Copying blob eb96ef222807 done
Copying config 42cdab313e done
Writing manifest to image destination
Storing signatures
toolbox: base image fedora-toolbox:30 resolved to registry.fedoraproject.org/f30/fedora-toolbox:30
toolbox: checking if container fedora-toolbox-30 already exists
toolbox: trying to create container fedora-toolbox-30
Created container: fedora-toolbox-30
Enter with: toolbox enter

$ toolbox -v enter
toolbox: resolved absolute path for /usr/bin/toolbox to /usr/bin/toolbox
toolbox: TOOLBOX_PATH is /usr/bin/toolbox
toolbox: checking if 'podman system migrate' exists
toolbox: migration not needed: 1.4.0 is unchanged
toolbox: Fedora generational core is f30
toolbox: base image is fedora-toolbox:30
toolbox: container is fedora-toolbox-30
toolbox: checking if container fedora-toolbox-30 exists
toolbox: trying to start container fedora-toolbox-30
toolbox: looking for /etc/profile.d/toolbox.sh in container fedora-toolbox-30
Error: unable to find user miabbott: no matching entries in passwd file
toolbox: copying /etc/profile.d/toolbox.sh to container fedora-toolbox-30
Error: error copying "/etc/profile.d/toolbox.sh" to "/var/home/miabbott/.local/share/containers/storage/overlay/1bc532e4d664fe242dc96aeebb585f24355b5faab77839aaf6fed0dde543398f/merged/etc/profile.d/toolbox.sh/toolbox.sh": mkdir /var/home/miabbott/.local/share/containers/storage/overlay/1bc532e4d664fe242dc96aeebb585f24355b5faab77839aaf6fed0dde543398f/merged/etc/profile.d/toolbox.sh: not a directory
toolbox: unable to copy /etc/profile.d/toolbox.sh to container fedora-toolbox-30
```

Creating new containers doesn't seem to help

**Comment by tpopela** [comment]
Could it be commit 0886fb466e7a73e7382255cf80c9459ffb1d84ac?

**Comment by debarshiray** [comment]
The key lines are:
```
toolbox: looking for /etc/profile.d/toolbox.sh in container fedora-toolbox-30
Error: unable to find user miabbott: no matching entries in passwd file
```

That's caused by a `podman exec --user "$USER" ... sh -c 'mount | grep ...'` invocation failing that's supposed to check if `/etc/profile.d/toolbox.sh` has already been bind mounted into the container.

**Comment by debarshiray** [comment]
While PR #202 side-steps the original issue, I'm still curious about what was going on.

What do these show:
```
$ podman stop fedora-toolbox-30
$ podman inspect --type container --format "{{.Path}} {{.Args}}" fedora-toolbox-30
$ podman start --attach fedora-toolbox-30
```

I am wondering if the user creation in the entry-point process failed for some reason.

**Comment by debarshiray** [comment]
Does the `podman start --attach` command above complain about:
```
failed to redirect /etc/localtime to /run/host/monitor/localtime
```

**Comment by debarshiray** [comment]
Is this better with the newly released toolbox-0.0.12 ?

**Comment by miabbott** [comment]
I'm using `toolbox-0.0.11-1.fc30.noarch` with `podman-1.4.4-3.fc30.x86_64` and I'm not hitting this anymore.  :+1:
